### PR TITLE
Make Linear Regression baseline optional for testing

### PR DESCRIPTION
**Pull request by siddharthnandy**

### CODE REVIEW

File: src/climate_learn/models/modules/forecast.py [hunk 1]
```diff
@@ -215,6 +215,134 @@ def test_step(self, batch: Any, batch_idx: int):
...
+                batch_size=len(x),
+            )
+
+        # rmse for linear regression baseline
+        # check if fit_lin_reg_baseline is called by checking whether self.lr_baseline is initialized
+        try:
+            lr_pred = self.lr_baseline.predict(
+                x.cpu().reshape((x.shape[0], -1))
+            ).reshape(y.shape)
+        except AttributeError as e:
```
Comment (jasonjewik) [inline src/climate_learn/models/modules/forecast.py:274]: I suspect the error is here. Lines 270-300 or so should be wrapped in an if-block.

**Review by jasonjewik — CHANGES_REQUESTED** [review]
I still get the following error when running `trainer.test`:
```
NotImplementedError: Expect climate_learn.models.fit_lin_reg_baseline be implemented before test steps.
```

File: src/climate_learn/models/modules/forecast.py (hunk 1 above)
Comment (siddharthnandy) [reply src/climate_learn/models/modules/forecast.py]: good catch, i made the change

**Review by jasonjewik — APPROVED** [review]
Minor error, but I fixed it. lgtm

**Pull request merged**